interface/main/messages/save.php

```diff
@@ -0,0 +1,246 @@
+        $HIPAA = ($_REQUEST['ME_hipaa_default_override'] ? $_REQUEST['ME_hipaa_default_override'] : '');
+        $MSGS = ($_REQUEST['MSGS_default_yes'] ? $_REQUEST['MSGS_default_yes'] : '');
+        $country_code = ($_REQUEST['PHONE_country_code'] ? $_REQUEST['PHONE_country_code'] : '1');
+
+        $myValues = array($facilities, $providers, $HIPAA, $country_code, $MSGS, $_REQUEST['POSTCARDS_local'], $_REQUEST['POSTCARDS_remote'], $_REQUEST['LABELS_local'], $_REQUEST['chart_label_type'], $_REQUEST['combine_time'], $_REQUEST['postcard_top']);
+
+        $_GLOBALS['chart_label_type'] = $_REQUEST['chart_label_type'];
+        sqlStatement('UPDATE `globals` SET gl_value = ? WHERE gl_name LIKE "chart_label_type" ', array($_REQUEST['chart_label_type']));
+
+        $result['output'] = sqlQuery($sql, $myValues);
+        if ($result['output'] == false) {
+            $result['success'] = "medex_prefs updated";
+        }
+        $result['logged_in'] = $MedEx->login();
+        $result['response'] = $MedEx->practice->sync($result['logged_in']['token']);
+        echo json_encode($result);
+    }
+    exit;
+}
+if ($_REQUEST['MedEx'] == "start") {
+    if (acl_check('admin', 'super')) {
+        $query = "SELECT * FROM users WHERE id = ?";
+        $user_data = sqlQuery($query, array($_SESSION['authUserID']));
+        $query = "SELECT * FROM facility WHERE primary_business_entity='1' LIMIT 1";
+        $facility = sqlFetchArray(sqlStatement($query));
+
+        $data['firstname'] = $user_data['fname'];
+        $data['lastname'] = $user_data['lname'];
+        $data['username'] = $_SESSION['authUser'];
+        $data['password'] = $_REQUEST['new_password'];
+        $data['email'] = $_REQUEST['new_email'];
+        $data['telephone'] = $facility['phone'];
+        $data['fax'] = $facility['fax'];
+        $data['company'] = $facility['name'];
+        $data['address_1'] = $facility['street'];
+        $data['city'] = $facility['city'];
+        $data['state'] = $facility['state'];
+        $data['postcode'] = $facility['postal_code'];
+        $data['country'] = $facility['country_code'];
+        $data['sender_name'] = $user_data['fname'] . " " . $user_data['lname'];
+        $data['sender_email'] = $facility['email'];
+        $data['callerid'] = $facility['phone'];
+        $data['MedEx'] = "1";
+        $data['ipaddress'] = $_SERVER['REMOTE_ADDR'];
+
+        $prefix = 'http://';
+        if ($_SERVER["SSL_TLS_SNI"]) {
+            $prefix = "https://";
+        }
+        $data['website_url'] = $prefix . $_SERVER['HTTP_HOST'] . $web_root;
+        $practice_logo = "$OE_SITE_DIR/images/practice_logo.gif";
+        if (!file_exists($practice_logo)) {
+            $data['logo_url'] = $prefix . $_SERVER['HTTP_HOST'] . $web_root . "/sites/" . $_SESSION["site_id"] . "/images/practice_logo.gif";
+        } else {
+            $data['logo_url'] = $prefix . $_SERVER['HTTP_HOST'] . $GLOBALS['images_static_relative'] . "/menu-logo.png";
+        }
```

Comment:
TODO
Seems odd that can't use a relative path for this. Will ensure this works on the demo farm (uses dockers which are not aware they are https); if it works, then this code block is good to go.